src/utils/decodedSourcemap.ts

```diff
@@ -1,12 +1,57 @@
-import { decode } from '@jridgewell/sourcemap-codec';
+import { decode, encode } from '@jridgewell/sourcemap-codec';
 import type {
+	DecodedSourceMapOrMissing,
 	ExistingDecodedSourceMap,
 	ExistingRawSourceMap,
 	SourceMapInput
 } from '../rollup/types';
 
 type Input = SourceMapInput | ExistingDecodedSourceMap | undefined;
 
+interface CachedSourcemapData {
+	encodedMappings: string | undefined;
+	decodedMappings: ExistingDecodedSourceMap['mappings'] | undefined;
+}
+
+const sourceMapCache = new WeakMap<ExistingDecodedSourceMap, CachedSourcemapData>();
```

Comment:
I'm using a `WeakMap` to manage the data behind the `decodedSourcemap` so that we can separately manipulate the state of those values without needing to modify the type of `ExistingDecodedSourceMap` or adding undocumented properties, etc.